src/tree.c

```diff
@@ -641,7 +643,11 @@ int git_tree__write_index(
 		git_index__set_ignore_case(index, false);
 	}
 
-	ret = write_tree(oid, repo, index, "", 0);
+	/* Grow the buffer beforehand to 1kb */
+	if ((ret = git_buf_grow(&shared_buf, 1024)) >= 0)
```

Comment:
In `git_treebuilder_write` I wanted to be a bit more conservative with the buffer allocation since it is not going to be reused. 

In `git_tree__write_index` I was going for more consistent performance by reducing the number of growths required. 

The previous buffer guess was `entrycount * 72`, so 1024 bytes is about 14 entries worth of buffer space. From there it would grow by 1.5x as needed.

I think leaving the initial estimate of `entrycount * 72` in `git_treebuilder_write_with_buffer` will in the worst case result in maybe 7 more resizes per merge, so I'm not too worried though. 

I could go either way, what do you think?